src/query/useFriendBot.ts

```diff
@@ -1,24 +1,26 @@
 import { useQuery } from "@tanstack/react-query";
+import { EmptyObj, Network } from "@/types/types";
 
 import { shortenStellarAddress } from "@/helpers/shortenStellarAddress";
 
 export const useFriendBot = ({
   network,
   publicKey,
 }: {
-  network: string;
+  network: Network | EmptyObj;
   publicKey: string;
 }) => {
-  const friendbotURL =
-    network === "futurenet"
-      ? "https://friendbot-futurenet.stellar.org"
-      : "https://friendbot.stellar.org";
-
   const query = useQuery({
     queryKey: ["friendBot"],
     queryFn: async () => {
+      if (!network.horizonUrl) {
+        throw new Error(`Please use a network that supports Horizon`);
+      }
+
       try {
-        const response = await fetch(`${friendbotURL}/?addr=${publicKey}`);
+        const response = await fetch(
+          `${network.horizonUrl}/friendbot?addr=${publicKey}`,
```

Comment:
I know we don't do it in the current Lab, but should we show a better error message when we have more info? For example, when funding an account that is already funded.

We get that info in the response
![image](https://github.com/stellar/laboratory/assets/7346473/b4d83939-9d20-431d-9902-b72884eef39e)